tests/integration/seccomp-notify.bats

```diff
@@ -0,0 +1,249 @@
+		.linux.seccomp = {
+			"defaultAction":"SCMP_ACT_ALLOW",
+			"listenerPath": "'"$SECCCOMP_AGENT_SOCKET"'",
+			"listenerMetadata": "bar",
+		}'
+
+	runc run test_busybox
+	[ "$status" -eq 0 ]
+}
+
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY empty listener path)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	scmp_act_notify_template "/bin/true" false '"mkdir"'
+	update_config '.linux.seccomp.listenerPath = ""'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+}
+
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY wrong listener path)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	scmp_act_notify_template "/bin/true" false '"mkdir"'
+	update_config '.linux.seccomp.listenerPath = "/some-non-existing-listener-path.sock"'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+}
+
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY abstract listener path)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	scmp_act_notify_template "/bin/true" false '"mkdir"'
+	update_config '.linux.seccomp.listenerPath = "@mysocketishere"'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+}
+
+# Check that killing the seccompagent doesn't block syscalls in
+# the container. They should return ENOSYS instead.
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY kill seccompagent)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	scmp_act_notify_template "sleep 4 && mkdir /dev/shm/foo" false '"mkdir"'
+
+	sleep 2 && teardown_seccompagent &
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+	[[ "$output" == *"Function not implemented"* ]]
+}
+
+# Check that starting with no seccomp agent running fails with a clear error.
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY no seccompagent)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	teardown_seccompagent
+
+	scmp_act_notify_template "/bin/true" false '"mkdir"'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+	[[ "$output" == *"failed to connect with seccomp agent"* ]]
+}
+
+# Check that agent-returned error for the syscall works.
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY error chmod)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	scmp_act_notify_template "touch /dev/shm/foo && chmod 777 /dev/shm/foo" false '"chmod", "fchmod", "fchmodat"'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+	[[ "$output" == *"No medium found"* ]]
+}
+
+# check that trying to use SCMP_ACT_NOTIFY with write() gives a meaningfull error.
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY write)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	scmp_act_notify_template "/bin/true" false '"write"'
+
+	runc run test_busybox
+	[ "$status" -ne 0 ]
+	[[ "$output" == *"SCMP_ACT_NOTIFY cannot be used for the write syscall"* ]]
+}
+
+# check that a startContainer hook doesn't get any extra file descriptor.
+@test "runc run [seccomp] (SCMP_ACT_NOTIFY startContainer hook)" {
+	requires_kernel 5.6
+	requires arch_x86_64
+
+	# shellcheck disable=SC2016
+	# For some (unknown) reason shellcheck doesn't like L199 together with the "if" in L210.
```

Comment:
Removing line numbers, thanks!

It seems the check ("SC2016: Expressions don't expand in single quotes, use double quotes for that.") is not really helpful when using `update_config()`. Is not just the startContainer command, there are several other issues that make this check and several good uses of `update_config()` incompatible:
* When using `update_config()` we use single quotes to properly delimit the $1 param to the function
* This shellcheck check checks that, basically, there are no `$` inside a single quote string. For example: it says that this part `"'"$SECCCOMP_AGENT_SOCKET"'"` will not escape the var, although it will as the opening single quote was in a previous line.
* If we are running commands in a subshell inside the single quoted string, like `$(cmd)` it also complains about that. I've tried to use a backticks for running commands, but it still fails the first complain in this list and even if I break the string to make shellcheck happy, it still complains with much useful clues about this line: `	update_config '.process.args = ["/bin/true"] | `. Which, honestly, couldn't find out why.

But I really think the SC2016 check is quite simple and not compatible with several valid uses of `update_config()`, like using single quotes to properly delimit the $1 param and if you grep the runc code for `disable=SC2016`, you will see that it is quite a lot out there, several times when using `update_config()`.

So, I'll add a comment saying that this checks fails for severa valid uses of update_config() where we use single quotes to delimit the $1 arg to `update_config()`. Let me know what you think.